app/controllers/comments_controller.rb

```diff
@@ -232,10 +232,34 @@ def unhide
     end
   end
 
+  def admin_delete
+    @comment = Comment.find(params[:comment_id])
+    authorize @comment
+    @comment.deleted = true
+
+    if @comment.save
+      redirect_url = @comment.commentable&.path
+      if redirect_url
+        flash[:success] = "Comment was successfully deleted."
+        redirect_to redirect_url
+      else
+        redirect_to_comment_path
+      end
+    else
+      redirect_to_comment_path
+    end
+  end
+
   private
 
   # Use callbacks to share common setup or constraints between actions.
   def set_comment
     @comment = Comment.find(params[:id])
   end
+
+  def redirect_to_comment_path
+    redirect_url = @comment.path
+    flash[:error] = "Something went wrong; Comment NOT deleted."
+    redirect_to "#{redirect_url}/mod"
```

Comment:
I think you can make this simpler by doing `redirect_to "#{@comment.path}/mod`. No need to make a one time variable.

Reply:
I did that initially Andy, but then I got a CodeClimate error about `Possible unprotected redirect`